packages/mui-joy/src/Link/Link.tsx

```diff
@@ -0,0 +1,235 @@
+import * as React from 'react';
+import styled from '../styles/styled';
+import useThemeProps from '../styles/useThemeProps';
+import Typography from '../Typography';
+import linkClasses, { getLinkUtilityClass } from './linkClasses';
+import { LinkProps, LinkTypeMap } from './LinkProps';
+
+const useUtilityClasses = (ownerState: LinkProps) => {
+  const { component, level, color, variant, underline, focusVisible, disabled } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      color && `color${capitalize(color)}`,
+      disabled && 'disabled',
+      focusVisible && 'focusVisible',
+      level,
+      underline && `underline${capitalize(underline)}`,
+      variant && `variant${capitalize(variant)}`,
+      component === 'button' && 'button',
+    ],
+  };
+
+  return composeClasses(slots, getLinkUtilityClass, {});
+};
+
+const LinkRoot = styled(Typography, {
+  name: 'MuiLink',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: LinkProps }>(({ theme, ownerState }) => {
+  return [
+    {
+      ...(ownerState.underline === 'none' && {
+        textDecoration: 'none',
+      }),
+      ...(ownerState.underline === 'hover' && {
+        textDecoration: 'none',
+        '&:hover': {
+          textDecoration: 'underline',
+        },
+      }),
+      ...(ownerState.underline === 'always' && {
+        textDecoration: 'underline',
+        '&:hover': {
+          textDecorationColor: 'inherit',
+        },
+      }),
+      // Same reset as ButtonBase.root
+      ...(ownerState.component === 'button' && {
+        position: 'relative',
+        WebkitTapHighlightColor: 'transparent',
+        backgroundColor: 'transparent', // Reset default value
+        // We disable the focus ring for mouse, touch and keyboard users.
+        outline: 0,
+        border: 0,
+        margin: 0, // Remove the margin in Safari
+        borderRadius: 0,
+        padding: 0, // Remove the padding in Firefox
+        ...(!!ownerState.variant && {
+          paddingInline: '0.25em', // better than left, right because it also works with writing mode.
+          marginInline: '-0.25em',
+        }),
+        cursor: 'pointer',
+        color: theme.vars.palette[ownerState.color!]?.textColor,
+        userSelect: 'none',
+        verticalAlign: 'middle',
+        MozAppearance: 'none', // Reset
+        WebkitAppearance: 'none', // Reset
+        '&::-moz-focus-inner': {
+          borderStyle: 'none', // Remove Firefox dotted outline.
+        },
+        [`&.${linkClasses.focusVisible}`]: theme.focus.default,
+      }),
+    },
+    ownerState.level && ownerState.level !== 'inherit' && theme.typography[ownerState.level],
+    theme.variants[ownerState.variant!]?.[ownerState.color!],
+    theme.variants[`${ownerState.variant!}Hover`]?.[ownerState.color!],
+    theme.variants[`${ownerState.variant!}Active`]?.[ownerState.color!],
+    theme.variants[`${ownerState.variant!}Disabled`]?.[ownerState.color!],
```

Comment:
```suggestion
    theme.variants[`${ownerState.variant || 'text'}Disabled`]?.[ownerState.color!],
```

for any `disabled` state, default to 'text'.